src/core.js

```diff
@@ -90,6 +90,8 @@ const stub = (genFunc, ...args) => {
   }
 }
 
+const MAX_STEPS = 1000
```

Comment:
When testing sagas with endless loop like `while(true) { ...` you can quite easily miss to provide exit `throw` in `envMapping`. This becomes even more challenging when you want to iterate a few times in the loop. Since we test the saga synchronously entering in the endless loop without exit condition is a something we should prevent and hence this `MAX_STEPS` comes into play. I feel that 1000 is reasonably high, also it can be changed per test with `opts.maxSteps`.